http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/model/LeonardoModel.scala

```diff
@@ -46,7 +46,7 @@ object ContainerRegistry {
 // Repo format: https://docs.docker.com/docker-hub/repos/
 // Docker hub doesn't have documentation about length limitation on repo name and image name, but 100 max seems reasonable
   final case object DockerHub extends ContainerRegistry {
-    def regex: Regex = """^([a-z0-9-_\/\S]){1,100}(:?[a-z0-9-_]{1,100}\S)$""".r
+    def regex: Regex = """^[A-Za-z0-9]+[\w./-]+(?::\w[\w.-]+)?(?:@[\w+.-]+:[A-Fa-f0-9]{32,})?$""".r
```

Comment:
can you add a test to https://github.com/DataBiosphere/leonardo/blob/develop/http/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/model/LeonardoModelSpec.scala#L217 in terms of what the previous regex is failing on

Reply:
I'm not sure this new regex is correct, it matches things like `"a//////////"`. I may clean it up (it relates to image autodetection).